hikari/impl/stateful_cache.py

```diff
@@ -63,6 +62,15 @@
 _ValueT = typing.TypeVar("_ValueT")
 
 
+def _copy_mapping(mapping: typing.Mapping[_KeyT, _ValueT]) -> typing.MutableMapping[_KeyT, _ValueT]:
+    # dict.copy ranges from between roughly 2 times to 5 times more efficient than casting to a dict so we want to
+    # try to use this where possible.
+    try:
+        return mapping.copy()  # type: ignore[attr-defined, no-any-return]
+    except (AttributeError, TypeError):
```

Comment:
When would this TypeError?

Reply:
if the "copy" method doesn't have the signature we're expecting here (assumption being made that if it has required args or kwargs then it's not the kind of copy method we were expecting)